src/actors/builtin/storage_power/storage_power_actor_code.go

```diff
@@ -189,6 +189,22 @@ func (a *StoragePowerActorCode_I) OnMinerSurprisePoStFailure(rt Runtime, numCons
 	}
 }
 
+func (a *StoragePowerActorCode_I) OnMinerEnrollCronEvent(rt Runtime, eventEpoch abi.ChainEpoch, sectorNumbers []sector.SectorNumber) {
+	rt.ValidateImmediateCallerAcceptAnyOfType(builtin.StorageMinerActorCodeID)
+	minerAddr := rt.ImmediateCaller()
+	minerEvent := &autil.MinerEvent_I{
+		MinerAddr_: minerAddr,
+		Sectors_:   sectorNumbers,
+	}
+
+	h, st := a.State(rt)
+	if _, found := st.Impl().CachedDeferredCronEvents_[eventEpoch]; !found {
+		st.Impl().CachedDeferredCronEvents_[eventEpoch] = autil.MinerEventSetHAMT_Empty()
```

Comment:
An array is probably sufficient for this per-miner list, no?

Reply:
Presumably we at least need some persistent data structure for this, since it will go on-chain. Do we have a persistent array implementation at the moment? If not, HAMT seems like the best choice.

Reply:
I think I had misread the context, this isn't per-miner so yes a persistent structure is appropriate. I think the AMT is more suited to lists but a minor issue we can probably resolve later.